kv/util/util.go

```diff
@@ -238,57 +321,65 @@ func watchAndUpdate(
 	getValue getValueFn,
 	update updateFn,
 	validate ValidateFn,
-	defaultValue interface{},
 	logger xlog.Logger,
-) (kv.ValueWatch, error) {
+) error {
 	if store == nil {
-		return nil, errNilStore
+		return errNilStore
 	}
 
-	watch, err := store.Watch(key)
+	var (
+		watch kv.ValueWatch
+		err   error
+	)
+
+	watch, err = store.Watch(key)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("could not establish initial watch: %v", err)
 	}
 
 	go func() {
 		for {
-			_, ok := <-watch.C()
-			if !ok {
-				return
+			if watch == nil {
```

Comment:
don't think you'll ever hit this case, the contract of the kv.ValueWatch is to fix itself once you get it. So users don't need to create it.

You could potentially add a error log in the !ok case, although that case should not never happen as well.

Reply:
Unless you feel strongly, I would actually prefer to leave it as is, and a comment that this case should never happen. My concern is that if we ever change the contract of `kv.ValueWatch` we'll change the behavior here as well unless we're careful.

Reply:
I'd prefer to rely on the Watch on doing this since that's the contract, the watch itself is doing the recreation more comprehensively, if the contract is broken then we have plenty places that will be broken as well. I think an error log is more than enough here.

Reply:
That's a good point and it would simplify the code here. Will update to add a comment on the contract of `kv.ValueWatch` and log an error.